guidance/hybridpath_guidance/hybridpath_guidance/hybridpath.py

```diff
@@ -0,0 +1,243 @@
+import numpy as np
+import matplotlib.pyplot as plt
+
+class HybridPathGenerator:
+    '''
+    Generates the coefficients for subpaths between given waypoints.
+    
+    Input data:
+    WP: Waypoints
+
+    r: Differentiability order
+
+    lambda_val: Curvature constant
+
+    plot_handle: Handle for the plot
+
+    Output data:
+    Path: A path structure with the following fields:
+        NumSubpaths: Number of subpaths
+        Order: Order of the subpaths
+        WP: Waypoints
+        LinSys: Linear system to solve the subpaths
+            A: Coefficients matrix
+            bx: x-vector for each subpath
+            by: y-vector for each subpath
+        coeff: Coefficients
+            a: x-coefficients
+            b: y-coefficients
+            a_der: x-coefficients for derivatives
+            b_der: y-coefficients for derivatives
+
+        Based on MATLAB code by prof. Roger Skjetne, NTNU
+    '''
+
+    def __init__(self, WP, r, lambda_val, plot_handle = None):
+        self.WP = WP
+        self.r = r
+        self.lambda_val = lambda_val
+        self.plot_handle = plot_handle
+        self.ord = 2*r + 1
+        self.Path = {
+            'NumSubpaths': len(WP) - 1,
+            'Order': self.ord,
+            'WP': {'x': WP[:,0], 'y': WP[:,1]},
+            'LinSys': {
+                'A': None,
+                'bx': [],
+                'by': []
+            },
+            'coeff': {
+                'a': [],
+                'b': [],
+                'a_der': [],
+                'b_der': []
+            }
+        }
```

Comment:
Whats the motivation for storing this as a dict instead of as a dataclass or related?